Prioritise consultations in the policy and engagement section

leenagupte opened the pull request:
Trello: https://trello.com/c/lZhCKzi7

### What's changed
Promote consultation content in the Policy and engagement section, and render them in a highlight box.

Example topic page: https://govuk-collections-pr-625.herokuapp.com/education/funding-and-finance-for-students

### Before
![screen shot 2018-04-18 at 18 08 33](https://user-images.githubusercontent.com/5793815/38947143-8905b5ec-4333-11e8-9fbc-80fa386b58c8.png)

### After
![screen shot 2018-04-18 at 18 09 26](https://user-images.githubusercontent.com/5793815/38947176-a75f2a14-4333-11e8-9e63-cf1ac3086455.png)

cbaines commented on app/presenters/supergroups/policy_and_engagement.rb:
```diff
@@ -8,6 +8,15 @@ def initialize
       @content = MostRecentContent.fetch(content_id: taxon_id, filter_content_purpose_supergroup: @name)
+      @content.select do |content_item|
+        consultation?(content_item.content_store_document_type)
+      end | @content
```
`|` is bitwise or in Ruby right?

leenagupte replied:
Yeah. I was trying to find a way to get consultations from the list of documents without losing the other documents. There's probably a nicer way to do this.

vanitabarrett commented on app/presenters/supergroups/supergroup.rb:
```diff
@@ -53,5 +53,9 @@ def document_list(taxon_id)
     end
+
+    def special_format_count(*)
+      0
```
Can we set '3' as the default, so we don't have to override this method in every supergroup class?

vanitabarrett commented on test/presenters/supergroups/policy_and_engagement_test.rb:
```diff
@@ -12,21 +12,177 @@
+      end
+
+      describe '#special_format_count' do
+        it 'only show applies special formatting to consultations' do
```
Not sure what you're trying to say in this test name... (or the one underneath)

leenagupte replied:
@vanitabarrett I'll rename this! That's some bad English right there!

vanitabarrett commented on test/presenters/supergroups/policy_and_engagement_test.rb:
```diff
@@ -12,21 +12,177 @@
+          assert_equal expected_result('open_consultation'), policy_and_engagement_supergroup.document_list(taxon_id)
+        end
+
+        it 'gets the closing date of future consultations' do
```
I'm a bit confused about the difference between this test and `gets the closing date of past consultations`. We seem to be testing both with a document type of `open_consultation`, but the test names suggest we're testing past (closed consultations) and future (open consultations)?

leenagupte replied:
@vanitabarrett The naming is bad. The description of the consultation closing date changes depending on whether it's in the past or future.

e.g. "Closing date 10 July 2018" vs "Date closed 10 July 2017"

In `gets the closing date of past consultations` the closing date is in 2017, and in `gets the closing date of future consultations` the date is in 2018.

leenagupte commented:
@vanitabarrett  I've replaced by #626 and addressed your comments there. On the request of @miaallers, I've also removed the "featured" flag from the highlight boxes in the new PR.
